tests/integration/conftest.py

```diff
@@ -93,22 +110,77 @@ async def app(
     # mypy doesn't see that ActiveStatus has a name
     await model.wait_for_idle(status=ActiveStatus.name)  # type: ignore
 
+    # test that the application is online
+    for _ in range(10):
+        try:
+            response = requests.get(
+                f"http://{await get_mattermost_ip(ops_test, application)}:8065",
+                timeout=5
+            )
+            if response.status_code == 200:
+                break
+        except:
+            pass
+        # wait a bit
+        time.sleep(10)
+
     yield application
 
 
 @pytest_asyncio.fixture(scope="module")
-async def mattermost_ip(
+async def test_entities(
     ops_test: OpsTest,
     app: Application,
 ):
-    """Get the IP address of the first unit of mattermost.
+    mattermost_ip = await get_mattermost_ip(ops_test, app)
 
-    Return the IP address of a mattermost unit.
-    """
-    unit_informations = json.loads(
-        (await ops_test.juju("show-unit", app.units[0].name, "--format", "json"))[1]
+    test_user = {"login_id": "test@test.test", "password": secrets.token_hex()}
+
+    # create a user
+    cmd = (
+        f"MMCTL_LOCAL_SOCKET_PATH=/tmp/mattermost.socket /mattermost/bin/mmctl"
+        f" --local user create --email {test_user['login_id']} --username test"
+        f" --password {test_user['password']}"
+    )
+    await ops_test.juju("run", "--application", app.name, cmd)
```

Comment:
Perhaps we discussed this before, in that case it could be worth documenting, what about using this library: https://github.com/Vaelor/python-mattermost-driver/blob/master/README.rst#installation